src/node_contextify.cc

```diff
@@ -279,6 +282,27 @@ void ContextifyContext::IsContext(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result.FromJust());
 }
 
+void ContextifyContext::GetStackSourceName(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsNumber());
+
+  uint32_t stack_depth = args[0].As<Integer>()->Value();
+
+  Local<StackTrace> stack =
+      StackTrace::CurrentStackTrace(env->isolate(),
+                                    stack_depth + 1,
+                                    StackTrace::kScriptName);
+
+  CHECK(stack_depth < stack->GetFrameCount());
```

Comment:
```suggestion
  CHECK_LT(stack_depth, stack->GetFrameCount());
```

Reply:
Thanks, I ended up reworking this section in the end to handle nested evaluations :)